### ISSUE TITLE
Redirect after auth for Kibana

### ISSUE THREAD

**Issue by jennyd** [ministryofjustice/cloud-platform#412]
## Service name

Kibana

## Service environment

- [ ] Dev / Development
- [ ] Staging
- [x] Prod / Production
- [ ] Other

## Impact on the service

I lose my carefully constructed Kibana queries 💎 

## Problem description

When I need to reauthenticate via Auth0 to access Kibana (which happens every few hours) I'm then redirected back to the Kibana homepage. Going back in the browser history doesn't get me back to the query I was last looking at, because the auth URL replaces that query URL in the history. Is it possible to redirect back to the actual last query URL after reauthenticating?

## Contact person

Jenny Duckett
@jennyd

**Comment by alkar** [comment]
I think we need to re-work (probably fork) this proxy. It would also allow us to do some rewriting too.

**Comment by pwyborn** [comment]
Hi Jenny - Just looking at the ticket “Redirect after auth for Kibana” - so far been unable to replicate the problem. Can I just ask does it happen all of the time (or only when you work remotely)?


jennyd [10:36 AM]
Yes, it’s always happened after I’ve left a tab open for a few hours and have to reauth
in the office too
eg leave this page open in a tab for a few hours and then go back to it: https://dsd-kibana.apps.cloud-platform-live-0.k8s.integration.dsd.io/_plugin/kibana/app/kibana#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:now-12h,mode:quick,to:now))&_a=(columns:!(message,%27@message%27),filters:!((%27$state%27:(store:appState),meta:(alias:!n,disabled:!f,index:f743b2c0-102a-11e8-8051-835cfbcbba58,key:%27@project%27,negate:!f,params:(query:pvb,type:phrase),type:phrase,value:pvb),query:(match:(%27@project%27:(query:pvb,type:phrase)))),(%27$state%27:(store:appState),meta:(alias:!n,disabled:!f,index:f743b2c0-102a-11e8-8051-835cfbcbba58,key:%27@environment%27,negate:!f,params:(query:prod,type:phrase),type:phrase,value:prod),query:(match:(%27@environment%27:(query:prod,type:phrase)))),(%27$state%27:(store:appState),meta:(alias:!n,disabled:!f,index:f743b2c0-102a-11e8-8051-835cfbcbba58,key:%27@log_type%27,negate:!f,params:(query:upstart,type:phrase),type:phrase,value:upstart),query:(match:(%27@log_type%27:(query:upstart,type:phrase))))),index:f743b2c0-102a-11e8-8051-835cfbcbba58,interval:auto,query:(language:lucene,query:%27%27),sort:!(%27@timestamp%27,desc))
after reauthing, I just get redirected back to a page that isn’t my query
and I think the auth popup page is replacing that URL in my browser history, so I can’t go back to my last query either

paulwyborn [10:45 AM]
Hi Jenny - next time it happens - could you perhaps do a screen dump of where it takes you to . Also does it do it no matter how you make up the query for instance by typing the actual query in or by using the filter?
Screen Shot 2018-10-16 at 10.44.01.png 


jennyd [10:46 AM]
yup, always happens

paulwyborn [10:46 AM]
or
Screen Shot 2018-10-16 at 10.44.16.png 


jennyd [10:47 AM]
I think it was trying to use the redirect_uri but it looked a bit mangled somehow - possibly there were two redirect_uri params in the URL
I’ll leave a tab open and get the URL later

paulwyborn [10:48 AM]
Ok Jenny thanks - let me know how it goes please

**Comment by pwyborn** [comment]
19/10/2018:
Jennyd [4:56 PM]
Just remembered about this, sorry! I’ve just been back to that tab which I opened on Tuesday at the URL above. Unusually, it didn’t decide to present me with the login box as soon as I went back to that page - I had to refresh it before it would do that. After I’d logged in again, it redirected me to https://dsd-kibana.apps.cloud-platform-live-0.k8s.integration.dsd.io/_plugin/kibana/app/kibana#/discover?_g=()&_a=(columns:!(_source),index:f743b2c0-102a-11e8-8051-835cfbcbba58,interval:auto,query:(language:lucene,query:''),sort:!('@timestamp',desc))

It looked like it went through another URL on the way, but redirected again before I could catch it

And if I go back in the browser history from that page I got to, the only other page in the history is the auth0 login page

22/10/2018
paulwyborn [12:31 PM]
Hi Jenny -thanks for that - will take a look - but general problems with github at the moment https://status.github.com/messages
